aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,458 @@
+     * in an {@link AuthException} and throw it on its calling thread.
+     * @throws AmplifyException the exception expected to be thrown when configuration fails.
+     * @throws JSONException has to be declared as part of creating a test JSON object
+     */
+    @Test(expected = AuthException.class)
+    public void testConfigureExceptionHandling() throws AmplifyException, JSONException {
+        JSONObject pluginConfig = new JSONObject().put("TestKey", "TestVal");
+        JSONObject json = new JSONObject().put("plugins",
+                new JSONObject().put(
+                        new AWSCognitoAuthPlugin().getPluginKey(),
+                        pluginConfig
+                )
+        );
+        AuthCategoryConfiguration authConfig = new AuthCategoryConfiguration();
+        authConfig.populateFromJSON(json);
+
+        doAnswer(invocation -> {
+            Callback<UserStateDetails> callback = invocation.getArgument(2);
+            callback.onError(new Exception());
+            return null;
+        }).when(mobileClient).initialize(any(), any(), any());
+
+        authCategory.configure(authConfig, getApplicationContext());
+    }
+
+    /**
+     * Tests that the signUp method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.signUp() with the username
+     * and password it received and, if included, the userAttributes and validationData sent in the options object.
+     * Also ensures that in the onResult case, the success callback receives a valid AuthSignUpResult.
+     * @throws AuthException test fails if this gets thrown since method should succeed
+     */
+    @Test
+    public void signUp() throws AuthException {
+        SignUpResult amcResult = new SignUpResult(
+            false,
+            new UserCodeDeliveryDetails(
+                    DESTINATION,
+                    DELIVERY_MEDIUM,
+                    ATTRIBUTE_NAME
+            ),
+            null
+        );
+
+        doAnswer(invocation -> {
+            assertEquals(USERNAME, invocation.getArgument(0));
+            assertEquals(PASSWORD, invocation.getArgument(1));
+            Map<String, String> attributeMap = invocation.getArgument(2);
+            assertEquals(ATTRIBUTE_VAL, attributeMap.get(ATTRIBUTE_KEY));
+            assertEquals(METADATA, invocation.getArgument(3));
+
+            Callback<SignUpResult> callback = invocation.getArgument(4);
+            callback.onResult(amcResult);
+            return null;
+        }).when(mobileClient).signUp(any(), any(), any(), any(), any());
+
+        AuthSignUpResult result = synchronousAuth.signUp(
+                USERNAME,
+                PASSWORD,
+                AWSCognitoAuthSignUpOptions.builder()
+                    .userAttributes(
+                        Collections.singletonList(new AuthUserAttribute(AuthUserAttributeKey.email(), ATTRIBUTE_VAL))
+                    )
```

Comment:
Feels like a lot of boilerplate for the user to add a dang attribute, no?

Can we achieve soemthing more like:

```
AWSCognitoAuthSignUpOptions.builder()
    .userAttribute(AuthUserAttributeKey.email(), ATTRIBUTE_VAL)
    .build();
```

Reply:
Yeah it is - the options object stores a list of attributes but I guess I could write some logic in the builder to handle taking a single attribute. Would just have to handle the case properly where someone uses both like `builder().userAttribute(...).userAttributes(...)`

Reply:
Haha so this was pretty neat - I'd made a slight error when refactoring this and my unit test caught it. :D